File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/dialogue/model/UserId.java
package com.study.webflux.rag.domain.dialogue.model;

import java.util.UUID;

public record UserId(
	String value
) {
	public UserId {
		if (value == null || value.isBlank()) {
			throw new IllegalArgumentException("userId cannot be null or blank");
		}
		if (value.length() > 128) {
			throw new IllegalArgumentException("userId too long");
		}
	}

	public static UserId of(String value) {
		return new UserId(value);
	}

	public static UserId generate() {
		return new UserId(UUID.randomUUID().toString());
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/dialogue/model/UserId.kt
package com.miyou.app.domain.dialogue.model

import java.util.UUID

data class UserId(
    val value: String,
) {
    init {
        require(value.isNotBlank()) { "userId cannot be null or blank" }
        require(value.length <= 128) { "userId too long" }
    }

    companion object {
        @JvmStatic
        fun of(value: String): UserId = UserId(value)

        @JvmStatic
        fun generate(): UserId = UserId(UUID.randomUUID().toString())
    }

    fun value(): String = value
}
